src/server/services/input-types.js

```diff
@@ -441,14 +441,71 @@ defineType({
     },
 });
 
+const TIME_MS = 1;
+const TIME_SEC = 1000 * TIME_MS;
+const TIME_MIN = 60 * TIME_SEC;
+const TIME_HR = 60 * TIME_MIN;
+const TIME_DAY = 24 * TIME_HR;
+const TIME_WEEK = 7 * TIME_DAY;
+const TIME_UNITS_MS = {
+    'ms': TIME_MS, 'msec': TIME_MS, 'msecs': TIME_MS, 'millisecond': TIME_MS, 'milliseconds': TIME_MS,
+    's': TIME_SEC, 'sec': TIME_SEC, 'secs': TIME_SEC, 'second': TIME_SEC, 'seconds': TIME_SEC,
+    'm': TIME_MIN, 'min': TIME_MIN, 'mins': TIME_MIN, 'minute': TIME_MIN, 'minutes': TIME_MIN,
+    'h': TIME_HR, 'hr': TIME_HR, 'hrs': TIME_HR,  'hour': TIME_HR, 'hours': TIME_HR,
+    'd': TIME_DAY, 'day': TIME_DAY, 'days': TIME_DAY,
+    'w': TIME_WEEK, 'week': TIME_WEEK, 'weeks': TIME_WEEK,
+};
+
 defineType({
     name: 'Date',
     description: 'A calendar date.',
-    baseType: 'Any',
+    baseType: 'String',
     parser: input => {
-        input = new Date(input);
-        if (isNaN(input.valueOf())) throw GENERIC_ERROR;
-        return input;
+        const grabNumber = input => {
+            const res = input.match(/^\s*([+-]?)\s*(\d+\.?\d*)\s*$/);
```

Comment:
Calling `trim` on the input around line 475 would allow you to remove the whitespace patterns at the beginning/end and simplify this a little bit.